### PR TITLE
Add MET, METplus packages

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR adds the MET and METplus packages to Spack.

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/met/package.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/met/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/packages/metplus/package.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/metplus/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepages of the two packages, the branches of each package exist and all of the version sha256.  Just asking that you correct the ending copyright year.

**Comment by AlexanderRichert-NOAA** [comment]
@tldahlgren is the package audit error a bug in `spack checksum --verfiy`? It appears to be trying to verify a version that's not in package.py; here's the last part of the debug output:
```
==> [2023-08-03-10:48:24.581379] SPIDER: [url=ParseResult(scheme='https', netloc='github.com', path='/dtcenter/MET/releases/', params='', query='', fragment='')]
Traceback (most recent call last):
  File "/n/AddMainSpackPackages/spack_met/bin/spack", line 52, in <module>
    sys.exit(main())
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack_installable/main.py", line 37, in main
    sys.exit(spack.main.main(argv))
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 1023, in main
    return _main(argv)
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 978, in _main
    return finish_parse_and_run(parser, cmd_name, env_format_error)
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 1006, in finish_parse_and_run
    return _invoke_command(command, parser, args, unknown)
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 647, in _invoke_command
    return_val = command(parser, args)
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/cmd/checksum.py", line 128, in checksum
    remote_versions = pkg.fetch_remote_versions()
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/package_base.py", line 2372, in fetch_remote_versions
    self.all_urls, self.list_url, self.list_depth, concurrency, reference_package=self
  File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/util/web.py", line 781, in find_versions_of_archive
    if url == reference_package.url_for_version(ver):
  File "/n/AddMainSpackPackages/spack_met/var/spack/repos/builtin/packages/met/package.py", line 79, in url_for_version
    else:
KeyError: '10.0.2'
```

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Comment by tldahlgren** [comment]
> @tldahlgren is the package audit error a bug in `spack checksum --verfiy`? It appears to be trying to verify a version that's not in package.py; here's the last part of the debug output:
> 
> ```
> ==> [2023-08-03-10:48:24.581379] SPIDER: [url=ParseResult(scheme='https', netloc='github.com', path='/dtcenter/MET/releases/', params='', query='', fragment='')]
> Traceback (most recent call last):
>   File "/n/AddMainSpackPackages/spack_met/bin/spack", line 52, in <module>
>     sys.exit(main())
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack_installable/main.py", line 37, in main
>     sys.exit(spack.main.main(argv))
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 1023, in main
>     return _main(argv)
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 978, in _main
>     return finish_parse_and_run(parser, cmd_name, env_format_error)
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 1006, in finish_parse_and_run
>     return _invoke_command(command, parser, args, unknown)
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/main.py", line 647, in _invoke_command
>     return_val = command(parser, args)
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/cmd/checksum.py", line 128, in checksum
>     remote_versions = pkg.fetch_remote_versions()
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/package_base.py", line 2372, in fetch_remote_versions
>     self.all_urls, self.list_url, self.list_depth, concurrency, reference_package=self
>   File "/n/AddMainSpackPackages/spack_met/lib/spack/spack/util/web.py", line 781, in find_versions_of_archive
>     if url == reference_package.url_for_version(ver):
>   File "/n/AddMainSpackPackages/spack_met/var/spack/repos/builtin/packages/met/package.py", line 79, in url_for_version
>     else:
> KeyError: '10.0.2'
> ```

Thanks for pointing this out.

**Comment by alecbcs** [comment]
@AlexanderRichert-NOAA just to give you a quick update, in automating the version checksum verification we'd forgotten that `spack checksum` currently doesn't handle infinite versions (`develop`) well. We're going to revert the CI change for the moment while we add support to the checksum command.

Once that reversion in merged into main, we can rebase your PR here and proceed as normal.

**Comment by AlexanderRichert-NOAA** [comment]
@alecbcs sounds good, thanks for the update!

**Pull request merged**